changelog/dmd.rvalue.dd

```diff
@@ -0,0 +1,22 @@
+Add primary expression of the form `__rvalue(expression)` which causes `expression` to be treated as an rvalue, even if it is an lvalue.
+
+Overloads on `ref`:
+```
+foo(S s); // selected if `s` is an rvalue
+foo(ref S s); // selected if argument `s` is an lvalue
+
+S s;
+S bar();
+...
+foo(s); // selects foo(ref S)
+foo(bar()); // selects foo(S)
+```
+With this change,
+```
+foo(__rvalue(s)); // selects foo(S)
+```
+This also applies to constructors and assignments, meaning move constructors and
+move assignments are enabled. Moving instead of copying can be much more resource
+efficient, as, say, a string can be moved rather than copied/deleted.
+
+A moved object is no longer valid after it is moved.
```

Comment:
I think this sentence is not correct... even if it is moved; it remains a valid object and it will be destroyed, so it must remain destroy-able. It just _may not be the SAME value_, that might have been taken, and some destructible default-state left in it's place.

There's further work we may be able to do to make that hand-over feel more like the sentence you wrote, but there are various choices and compromises (we discussed numerous options dconf), but that wasn't even remotely settled. For the time being, you should remove this final sentence.

Reply:
I know we did not precisely define this behavior in our discussions, and this PR doesn't address those issues, either. That is best left to a followup PR, as it is orthogonal to this PR, and this is too big already. The bigger it is, the more fragile it is, the harder to do a merge, the harder to understand it, etc.

Reply:
For the moment, saying it's invalid is ok.

Reply:
But it's not invalid; it is very valid, and it will be destroyed.
Please delete this sentence...

Reply:
Or, remove the changelog entry entirely; which was my initial suggestion...